src/python_routing_v02/compute_nhd_routing_SingleSeg.py

```diff
@@ -213,21 +213,15 @@ def compute_mc_reach_up2down(
         writeString = f"\nreach: {head_segment} (order: {reach['seqorder']} n_segs: {len(reach['segments'])}  isterminal: {reach['upstream_reaches'] == {supernetwork_data['terminal_code']}} )  reach tail: {reach['reach_tail']}  upstream seg : "
 
     # upstream flow per reach
-    qup_tmp = 0
-    quc_tmp = 0
+    qup = 0.0
+    quc = 0.0
     # import pdb; pdb.set_trace()
-    if reach["upstream_reaches"] == {supernetwork_data["terminal_code"]}:  # Headwaters
-        qup_tmp = 0.0  # no flows in a head channel, only laterals
-        quc_tmp = qup_tmp  # no flows in a head channel, only laterals
-    else:  # Loop over upstream reaches
+    if reach["upstream_reaches"] != {supernetwork_data["terminal_code"]}:  # Not Headwaters
         for us in connections[reach["reach_head"]]["upstreams"]:
             if write_output:
                 writeString = writeString + f"\n upstream seg : {us}"
-            qup_tmp += flowdepthvel[us]["flow"]["curr"]
-            if assume_short_ts:
-                quc_tmp = qup_tmp
-            else:
-                quc_tmp += flowdepthvel[us]["flow"]["curr"]
+            qup += flowdepthvel[us]["flow"]["prev"]
+            quc += flowdepthvel[us]['flow']['curr']
     if write_output:
         writetoFile(file, writeString)
 
```

Comment:
breaks with '-o' flag because of qup_tmp in write statement on line 234 in current version.